### PR TITLE
Create get started page for OpenShift in VSCode Open walkthrough

### PR DESCRIPTION
**Pull request by msivasubramaniaan**
Added Walkthrough steps for the extension

Ref: https://github.com/redhat-developer/vscode-openshift-tools/issues/2414

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1561,6 +1556,69 @@
 				}
 			]
 		},
+        "walkthroughs": [
+            {
+                "id": "openshiftWalkthrough",
+                "title": "Getting Started with OpenShift",
+                "description": "A sample walkthrough",
```
Comment (mohitsuman) [inline package.json:1563]: Start your application deployment on OpenShift or Kubernetes

File: package.json [hunk 2]
```diff
@@ -1561,6 +1556,69 @@
...
+                "steps": [
+                    {
+                        "id": "login",
+                        "title": "Login to Openshift",
+                        "description": "This step will helps with login to Openshift.\n[Connect using remote OpenShift Cluster](command:openshift.explorer.login)\n[Connect using OpenShift Local](command:openshift.explorer.addCluster?[\"crc\"])\n[Connect using Developer Sandbox](command:openshift.explorer.addCluster?[\"sandbox\"])",
```
Comment (mohitsuman) [inline package.json:1568]:
```suggestion
                        "description": "Please select a way to login to an Openshift instance.\n[Connect using remote OpenShift Cluster](command:openshift.explorer.login)\n[Connect using OpenShift Local](command:openshift.explorer.addCluster?[\"crc\"])\n[Connect using Developer Sandbox](command:openshift.explorer.addCluster?[\"sandbox\"])",
```

File: package.json [hunk 3]
```diff
@@ -1561,6 +1556,69 @@
...
+                        "description": "This step will push the component into the cluster.",
+                        "media": {
+                            "image": "images/walkthrough/push.gif",
+                            "altText": "push"
+                        }
+                    },
+                    {
+                        "id": "debug",
+                        "title": "Debug the Component locally",
+                        "description": "This step helps to understand, debugging a component locally.",
```
Comment (mohitsuman) [inline package.json:1604]:
```suggestion
                        "description": "Debug and test remote applications deployed from your IDE to OpenShift/Kubernetes. No more having to exit your IDE to push your application.",
```

File: package.json [hunk 4]
```diff
@@ -1561,6 +1556,69 @@
...
+                            "altText": "push"
+                        }
+                    },
+                    {
+                        "id": "openInBrowser",
+                        "title": "Open the Component in Browser",
+                        "description": "This step helps to understand, opening the component in browser.",
+                        "media": {
+                            "image": "images/walkthrough/push.gif",
+                            "altText": "push"
```
Comment (mohitsuman) [inline package.json]: Change the alt text and the image gif.

File: package.json [hunk 5]
```diff
@@ -1561,6 +1556,69 @@
...
+                            "onCommand:openshift.explorer.login",
+                            "onCommand:openshift.explorer.addCluster",
+                            "onCommand:openshift.explorer.login.credentialsLogin",
+                            "onCommand:openshift.explorer.login.tokenLogin"
+                        ]
+                    },
+                    {
+                        "id": "createComponent",
+                        "title": "Create a Component using Registry Viewer",
+                        "description": "This step will run a command and check off once it has been run.\n[Create Component](command:openshift.componentType.newComponent)",
```
Comment (mohitsuman) [inline package.json:1595]:
```suggestion
                        "description": Create a component in the current working directory with the file `devfile.yaml` from the devfile registry selected. Devfile is a manifest file that contains information about various resources (URL, Storage, Services, etc.) that correspond to your component. \n[Create Component](command:openshift.componentType.newComponent)",
```

File: package.json [hunk 6]
```diff
@@ -1561,6 +1556,69 @@
...
+                            "altText": "create component"
+                        },
+                        "completionEvents": [
+                            "onCommand:openshift.componentType.newComponent"
+                        ]
+                    },
+                    {
+                        "id": "pushComponent",
+                        "title": "Push a Component to the connected cluster",
+                        "description": "This step will push the component into the cluster.",
```
Comment (mohitsuman) [inline package.json:1607]:
```suggestion
                        "description": "The component created above is available locally. To deploy the component on the connect OpenShift/Kubernetes cluster, we need to `Push` the Component",
```

**Review by mohitsuman — CHANGES_REQUESTED** [review]
@msivasubramaniaan Please update the descriptions accordingly as per the suggestions. We also need to update with the zoom in gifs for every step.

File: package.json [hunk 7]
```diff
@@ -1561,6 +1557,81 @@
 				}
 			]
 		},
+        "walkthroughs": [
+            {
+                "id": "openshiftWalkthrough",
+                "title": "Getting Started with OpenShift",
+                "description": "Start your application deployment on OpenShift or Kubernetes",
```
Comment (mohitsuman) [inline package.json:1564]:
```suggestion
                "description": "Start your application development on OpenShift or Kubernetes",
```

File: package.json [hunk 8]
```diff
@@ -1561,6 +1557,81 @@
...
+                "steps": [
+                    {
+                        "id": "login",
+                        "title": "Login to OpenShift instance",
+                        "description": "Please select a way to login to an Openshift instance.\n[Connect using remote OpenShift Cluster](command:openshift.explorer.login)\n[Connect using OpenShift Local](command:openshift.explorer.addCluster?[\"crc\"])\n[Connect using Developer Sandbox](command:openshift.explorer.addCluster?[\"sandbox\"])",
```
Comment (mohitsuman) [inline package.json:1569]:
```suggestion
                        "description": "Please select a way to login to an OpenShift instance.\n[Connect using remote OpenShift Cluster](command:openshift.explorer.login)\n[Connect using OpenShift Local](command:openshift.explorer.addCluster?[\"crc\"])\n[Connect using Red Hat Developer Sandbox](command:openshift.explorer.addCluster?[\"sandbox\"])",
```

File: package.json [hunk 9]
```diff
@@ -1561,6 +1557,81 @@
...
+                        "completionEvents": [
+                            "onCommand:openshift.explorer.login",
+                            "onCommand:openshift.explorer.addCluster",
+                            "onCommand:openshift.explorer.login.credentialsLogin",
+                            "onCommand:openshift.explorer.login.tokenLogin"
+                        ]
+                    },
+                    {
+                        "id": "showRegistries",
+                        "title": "Show all the registries",
```
Comment (mohitsuman) [inline package.json:1583]:
```suggestion
                        "title": "View the default and custom devfile registries.",
```

**Comment by mohitsuman** [comment]
@msivasubramaniaan the gifs looks good to me. Just commit the suggested changes.

**Review by mohitsuman — APPROVED** [review]
LGTM

File: package.json [hunk 10]
```diff
@@ -1561,6 +1557,81 @@
...
 			]
 		},
+        "walkthroughs": [
+            {
+                "id": "openshiftWalkthrough",
+                "title": "Getting Started with OpenShift",
+                "description": "Start your application development on OpenShift or Kubernetes",
+                "steps": [
+                    {
+                        "id": "login",
```
Comment (dgolovin) [inline package.json:1567]: image

This step does not look right to me. Login is just a way to get access to cluster resources and it does not matter how cluster is provisioned or where it is located.

What I see is attempt to mix cluster login, provisioning and location in one step.

I suggest to bubble up one more step. 'Get access to OpenShift cluster' or 'Provision or use existing OpenShift cluster'
Where we can have button to open 'Add Cluster Editor'. Then reduce Login step to just a description and login button.

**Review by dgolovin — CHANGES_REQUESTED** [review]

File: package.json (hunk 10 above)
Comment (msivasubramaniaan) [reply package.json:1567]: image
Changed the login title and description. Please refer the screenshot

Comment (mohitsuman) [reply package.json:1567]: I think this covers the use case of differentiating the Login and provision workflow. The description also works for me. LGTM.

Comment (dgolovin) [reply package.json:1567]: Well, it is still two questions answered in one step.

'Connect using OpenShift Local' sounds awkward to me, I would say 'Connect using ssh' or 'Connect to OpenShift using SSH'. Both sounds good to me. Lets try 'Connect to OpenShift Using OpenShift Local' does that sound good? It is not for me.

IMHO first someone who is interested would ask where/how one can get OpenShift instance? Then ask how to work with it or see what is in it through the UI elements provided by extension they've just installed and that would require to log into the cluster.

**Review by dgolovin — CHANGES_REQUESTED** [review]
Please, see my comment above.

File: package.json (hunk 10 above)
Comment (mohitsuman) [reply package.json:1567]: Here OpenShift local refers to renamed CodeReadyContainers. So connect using OpenShift local should open the CRC view we currently have.

**Review by dgolovin — APPROVED** [review]

**Pull request merged**